src/abstract-interpretation/data-frame/domain.ts

```diff
@@ -26,6 +29,12 @@ export const DataFrameTop: DataFrameDomain = {
 	rows:     IntervalTop
 };
 
+export type DataFrameStateDomain = Map<NodeId, DataFrameDomain>;
+
+export function equalColNames(X1: ColNamesDomain, X2: ColNamesDomain): boolean {
+	return X1 === X2 || (X1 !== ColNamesTop && setEquals(new Set(X1), new Set(X2)));
+}
+
 export function leqColNames(X1: ColNamesDomain, X2: ColNamesDomain): boolean {
```

Comment:
I think there is a merit in generalizing these operations so that they are not specific to the colnames domain - given that they just represent the exact set domain